cli.js

```diff
@@ -51,15 +61,14 @@ if (cmd === 'init') {
 
 	const name = argv._[1]
 	if ('string' !== typeof name || !name) showError('Missing name.')
-	// todo: use get-port if no port given
-	const port = parseInt(argv._[2])
-	if (Number.isNaN(port) || port <= 0) showError('Invalid or missing port.')
 
-	const id = randomBytes(8).toString('hex')
+	ensurePort(parseInt(argv._[2])).then(port => {
+		const id = randomBytes(8).toString('hex')
 
-	mkdirp.sync(path.dirname(cfgPath))
-	fs.writeFileSync(cfgPath, JSON.stringify({id, name, port}))
-	console.info('Done.')
+		mkdirp.sync(path.dirname(cfgPath))
```

Comment:
Don't use sync code inside async code. Either move the `mkdirp` call up or use `new Promise(…)`.

Reply:
Moved mkdirp up